### PR TITLE
kubepkg: Restore kubernetes-cni package templates

### PR DESCRIPTION
**Pull request by justaugustus**
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Follow-up to #1379 and https://github.com/kubernetes/release/pull/1375.
ref: https://github.com/kubernetes/kubernetes/issues/92242

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

@saschagrunert -- This is what I was referring to in https://github.com/kubernetes/release/pull/1379#discussion_r446083716:

> You'll need to restore variable references in the package templates as well. They should roughly match what was removed in #1330.

#### Does this PR introduce a user-facing change?

```release-note
kubepkg: Restore kubernetes-cni package templates
```

### CODE REVIEW

File: pkg/kubepkg/kubepkg.go [hunk 1]
```diff
@@ -600,11 +605,14 @@ func GetDependencies(packageDef *PackageDefinition) (map[string]string, error) {
...
-	if packageDef.Name == "kubeadm" {
+	switch packageDef.Name {
+	case "kubelet":
+		deps["kubernetes-cni"] = MinimumCNIVersion
+	case "kubeadm":
 		deps["kubelet"] = minimumKubernetesVersion
 		deps["kubectl"] = minimumKubernetesVersion
-		deps["cri-tools"] = minimumCRIToolsVersion
 		deps["kubernetes-cni"] = MinimumCNIVersion // deb based kubeadm still requires kubernetes-cni
+		deps["cri-tools"] = minimumCRIToolsVersion
```
Comment (saschagrunert) [inline pkg/kubepkg/kubepkg.go:615]: Nit: moving this line seems unnecessary

**Comment by justaugustus** [comment]
Thanks @saschagrunert!

The test is [still failing](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-release-build-packages-debs/1277289725140406272):

```shell
 curl --fail -sSL --retry 5 \
	-o usr/bin/kubelet \
	"https://dl.k8s.io/ci/v1.19.0-beta.2.511+aadaa5d6a917fc/bin/linux/arm/kubelet"
dpkg-architecture: warning: specified GNU system type arm-linux-gnueabihf does not match CC system type x86_64-linux-gnu, try setting a correct CC environment variable
 dpkg-source --before-build .
 fakeroot debian/rules clean
 debian/rules build
 fakeroot debian/rules binary
curl: (22) The requested URL returned error: 404 
make: *** [debian/rules:12: binary] Error 22
dpkg-buildpackage: error: fakeroot debian/rules binary subprocess returned exit status 2
level=fatal msg="running debian package build: command /usr/bin/dpkg-buildpackage --unsigned-source --unsigned-changes --build=binary --host-arch armhf did not succeed: dpkg-architecture: warning: specified GNU system type arm-linux-gnueabihf does not match CC system type x86_64-linux-gnu, try setting a correct CC environment variable\n dpkg-source --before-build .\n fakeroot debian/rules clean\n debian/rules build\n fakeroot debian/rules binary\ncurl: (22) The requested URL returned error: 404 \nmake: *** [debian/rules:12: binary] Error 22\ndpkg-buildpackage: error: fakeroot debian/rules binary subprocess returned exit status 2\n"
```

The good news is that's a new failure, so we're getting closer.

**Comment by justaugustus** [comment]
Ahhh, figured it out, I think.
The CI download is looking for a cross-build, which latest.txt is not guaranteed to be.

Working on a fix now.

**Comment by saschagrunert** [comment]
> Ahhh, figured it out, I think.
> The CI download is looking for a cross-build, which latest.txt is not guaranteed to be.
> 
> Working on a fix now.

Ah cool, thank you! Let me know if I can support somehow. I guess we could also add those tests as presubmits to get earlier feedback.

**Comment by justaugustus** [comment]
> I guess we could also add those tests as presubmits to get earlier feedback.

Yep, that was the goal eventually, but at the time the tool was a bit rougher around the edges! :sweat_smile:

**Pull request merged**